ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -220,6 +220,10 @@ impl Blockstore {
             & self
                 .db
                 .delete_range_cf::<cf::OptimisticSlots>(&mut write_batch, from_slot, to_slot)
+                .is_ok()
+            & self
```

Comment:
Why are we doing bit-and here?!
Shouldn't these all be `&&`?

Reply:
did some sleuthing, looks like it's intentional to avoid short circuiting https://github.com/solana-labs/solana/pull/7526

Reply:
Yep, confirmed that `&` is the desired behavior here. No short circuit so try to purge as much as possible; logical `AND` to see / report if any of them failed